### PR TITLE
Bookstore CMS: Use Redux in React component

### PR DESCRIPTION
**Pull request by Profsain**
- Update the books state
- connect redux to react component
- Added addbook redux functionality
- Added remove button dispatch
- Added check status dispatch
-
screencapture-localhost-3000-2022-07-19-15_55_05

### CODE REVIEW

File: src/components/Bookcard.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { removeBook } from '../redux/books/books';
 import '../styles/Bookcard.css';
 
 export default function Bookcard({ title, author }) {
+  const dispatch = useDispatch();
+  const removeBookHandler = () => {
+    dispatch(removeBook());
```
Comment (topeogunleye) [inline src/components/Bookcard.js]: - [ ] Your removeBook function in  `../redux/books/books` is expecting an id as a parameter, but you didn't add it here.  Please kindly add your id parameter to ensure your remove book functionality is working.

File: src/components/Bookcard.js [hunk 2]
```diff
@@ -11,7 +18,12 @@ export default function Bookcard({ title, author }) {
         <p>{author}</p>
         <div className="Buttons">
           <button type="button">Comment</button>
-          <button type="button">Remove</button>
+          <button
+            type="button"
+            onClick={removeBookHandler}
```
Comment (topeogunleye) [inline src/components/Bookcard.js:25]: - [ ] Kindly ensure that your removeBookHandler function takes the id from the book data and dispatches the `removeBook` action creator with the id as its argument

File: src/components/Categories.js [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
   return (
     <div>
-      <Header />
-      <button type="button">Check status</button>
+      <button
+        type="button"
+        onClick={() => dispatch(checkBookStatus())}
+      >
+        Check status
+      </button>
```
Comment (topeogunleye) [inline src/components/Categories.js:16]: - [ ] I kindly suggest that you display the return statement of 'Under Construction' from the `checkBookStatus` action when the Category component loads.

**Review by topeogunleye — CHANGES_REQUESTED** [review]
> # Changes Required 
> ### Hi @Profsain,
> Nice job, you are almost there.
> 
> 
> 
> ## (Highlights) Good Points: 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good summary description :heavy_check_mark:
>
> ## (Changes Required) Aspects to improve: 
> * [ ]  Please check my inline comments.
> 
> ## [Optional] suggestions:
> * Nothing to mention.
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> **Cheers and Happy coding! :clap::clap::clap:**
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear
> Please, remember to tag @topeogunleye in your question so I can receive the notification.
>
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by rustam-tolipov — APPROVED** [review]
## Hello @Profsain👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**